docs/developer-guide/views-examples.md

```diff
@@ -0,0 +1,163 @@
+---
+description: Learn to extend a View in Lite XL for extra functionality.
+---
+
+## Example: ToolbarView
+
+![Screenshot of a ToolBarView example][screenshot-toolbarview]
+
+### Creating a plugin
+
+Before reading further, it may be useful to refresh Lua basics by reading the [Lua manual][learning-lua].
+
+### Adding configuration settings
+
+WIP: write some setting option examples
+
+WIP: explain why we merge the tables
+
+```lua
+local toolbar = common.merge({
+  -- plugin configuration options go here
+}, config.plugins.toolbar)
+```
+
+### Checking if the plugin directory exists
+
+The code below is used to check if every folder detected by `get_file_info()` 
+actually exists and if it doesn't, a `nil` value is returned.
+
+```lua
+local function get_plugin_directory()
+  local paths = {
+    USERDIR .. PATHSEP .. "plugins" .. PATHSEP .. "toolbar",
+    DATADIR .. PATHSEP .. "plugins" .. PATHSEP .. "toolbar"
+  }
+  for i, v in ipairs(paths) do
+    if system.get_file_info(v) then
+      return v
+    end
+  end
+  return nil
+end
+```
+
+### Creating the view
+
+The code below creates an instance of our custom Toolbar View, inherits the ToolbarView super-constructor, 
+specifies the desired icon font and assigns the icons to their respective commands.
+
+```lua
+local ToolBar = ToolbarView:extend()
+
+function ToolBar:new()
+  ToolBar.super.new(self)
+  self.toolbar_font = renderer.font.load(get_plugin_directory() .. PATHSEP .. "toolbar.ttf", style.icon_big_font:get_size())
+  self.toolbar_commands = {
+    {symbol = "", command = "core:open-log"},
+    {symbol = "", command = "core:new-doc"},
+    {symbol = "", command = ""},
+    {symbol = "", command = ""},
+    {symbol = "", command = ""},
+    {symbol = "", command = ""},
+  }
+end
+```
+
+`local ToolBar = ToolbarView:extend()` is used to create a new custom View and make it 
+inherit ToolbarView.
+
+The `renderer.font.load` function assigns the icon font to the View.
+
+The `get_plugin_directory()` function gets the plugin folder path.
+
+### Adding the view
+
+```lua
+toolbar.example_toolbar_view = ToolBar()
+toolbar.example_toolbar_node = TreeView.node.b:split("up", toolbar.example_toolbar_view, {y = true})
+return toolbar
+```
+
+The first line of code saves the toolbar in an exported table.
```

Comment:
You don't have to explain the code line by line. Here you can specifically explain what the second line does.

For the explanation of the first and third, you can move it above the codeblock